### PR TITLE
[ Fixes: #2 ] Training and Testing a Classification Model- Linear Model (Stochastic Gradient Descent) on winequality.csv 

### LINKED ISSUE #2: [Outreachy applications] Startup task: Train and test a classification model
This is a good way to get started with the environment and the problem domain. It will also provide the basis for a test case for future work. At a minimum, you should:

- load a dataset from the repo
- train a classification model from [scikit-learn](https://scikit-learn.org/stable/supervised_learning.html)
- compute an [evaluation metric](https://scikit-learn.org/stable/modules/model_evaluation.html) on a held-out test set

Feel free to include any additional steps you feel are relevant or you are interested in trying out, such as:

- basic exploratory analysis of the dataset
- data preprocessing
- hyperparameter tuning

When you start work on this task, please post a comment here indicating which dataset and model you will be working with so that other contributors can avoid duplicating your work.

### PR DESCRIPTION
**Pull request by archisha-chandel**
[ Fixes #2 ]
Linear Model- Stochastic Gradient Descent under 'sklearn' has been implemented on winequality.csv
It includes the following:

- Exploratory data analysis

- Data pre-processing

- Hyper-parameter tuning of the model (done manually through experimentation)

- Fitting and Prediction on train-test data

- Computing evaluating metrics

The classification report looks as follows:
image

`pandas_profiling` is an extra library that has been added to the environment and it is of great advantage which are stated below:

_For each column the following statistics - if relevant for the column type - are presented in an interactive HTML report:_

1. **Type inference**: detect the types of columns in a dataframe.
2. **Essentials**: type, unique values, missing values
3. **Quantile statistics** like minimum value, Q1, median, Q3, maximum, range, interquartile range
4. **Descriptive statistics** like mean, mode, standard deviation, sum, median absolute deviation, coefficient of variation, kurtosis, skewness
5. **Most frequent values**
6. **Histogram**
7. **Correlations** highlighting of highly correlated variables, Spearman, Pearson and Kendall matrices
8. **Missing values** matrix, count, heatmap and dendrogram of missing values
9. **Text analysis learn** about categories (Uppercase, Space), scripts (Latin, Cyrillic) and blocks (ASCII) of text data.

### CODE REVIEW

**Comment by archisha-chandel** [comment]
@mlopatka you are right about `quality` and `recommend` being correlated but `quality` is used as a feature. According to what I know, if two features have high correlation we drop one of them but if a feature and target have correlation it is considered profitable for the model. Correct me if I am wrong.

Although, I did drop `quality` column as requested and made the corresponding changes as required.
